opal-core/src/main/java/org/obiba/opal/core/service/AnalysisExportServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+package org.obiba.opal.core.service;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.sun.istack.NotNull;
+import com.sun.istack.Nullable;
+import org.json.JSONObject;
+import org.obiba.opal.core.domain.OpalAnalysis;
+import org.obiba.opal.core.domain.OpalAnalysisResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+@Component
+public class AnalysisExportServiceImpl implements AnalysisExportService {
+
+  private static final Logger log = LoggerFactory.getLogger(AnalysisExportServiceImpl.class);
+
+  @Autowired
+  private OpalAnalysisService opalAnalysisService;
+
+  @Autowired
+  private OpalAnalysisResultService opalAnalysisResultService;
+
+
+  @Override
+  public void exportProjectAnalysis(@NotNull String projectName, OutputStream outputStream) throws IOException {
+    exportProjectAnalysis(projectName, outputStream, null);
+  }
+
+  @Override
+  public void exportProjectAnalysis(@NotNull String projectName, @NotNull OutputStream outputStream, @Nullable String tableName) throws IOException {
+    Assert.isTrue(!Strings.isNullOrEmpty(projectName), "Project name cannot be empty or null.");
+    Assert.notNull(outputStream, "outputStream cannot be null.");
+
+    List<OpalAnalysis> analyses = Strings.isNullOrEmpty(tableName)
+      ? Lists.newArrayList(opalAnalysisService.getAnalysesByDatasource(projectName))
+      : Lists.newArrayList(opalAnalysisService.getAnalysesByDatasourceAndTable(projectName, tableName));
+
+    ZipOutputStream zipOutputStream = new ZipOutputStream(outputStream);
+
+    for (OpalAnalysis analyse : analyses) {
+      String analysisId = analyse.getId();
+
+      zipOutputStream.putNextEntry(new ZipEntry(Paths.get("analyses", analysisId, "analysis.json").toString()));
+      zipOutputStream.write(new JSONObject(analyse).toString().getBytes());
+      zipOutputStream.closeEntry();
+
+      for (OpalAnalysisResult result : opalAnalysisResultService.getAnalysisResults(analysisId)) {
```

Comment:
Shouldn't it get the latest result only? Or have a parameter that allows to get all, but default is only the latest.